packages/host/app/services/realm-server.ts

```diff
@@ -0,0 +1,190 @@
+import Service from '@ember/service';
+import { service } from '@ember/service';
+
+import { task, restartableTask, rawTimeout } from 'ember-concurrency';
+
+import window from 'ember-window-mock';
+
+import { SupportedMimeType } from '@cardstack/runtime-common';
+import { RealmAuthClient } from '@cardstack/runtime-common/realm-auth-client';
+
+import type LoaderService from './loader-service';
+import type { ExtendedClient } from './matrix-sdk-loader';
+
+interface RealmServerTokenClaims {
+  user: string;
+}
+
+// iat - issued at (seconds since epoch)
+// exp - expires at (seconds since epoch)
+type RealmServerJWTPayload = RealmServerTokenClaims & {
+  iat: number;
+  exp: number;
+};
+
+type AuthStatus =
+  | { type: 'logged-in'; token: string; claims: RealmServerJWTPayload }
+  | { type: 'anonymous' };
+
+export default class RealmServerService extends Service {
+  @service declare loaderService: LoaderService;
+  private auth: AuthStatus = { type: 'anonymous' };
+  private client: ExtendedClient | undefined;
+
+  setClient(client: ExtendedClient) {
+    this.client = client;
+    this.token =
+      window.localStorage.getItem(sessionLocalStorageKey) ?? undefined;
+  }
+
+  async createRealm(args: {
+    endpoint: string;
+    name: string;
+    iconURL?: string;
+    backgroundURL?: string;
+  }) {
+    if (!this.client) {
+      throw new Error(`Cannot create realm without matrix client`);
+    }
+    await this.login();
+    if (this.auth.type !== 'logged-in') {
+      throw new Error('Could not login to realm server');
+    }
+
+    let response = await this.loaderService.loader.fetch(
```

Comment:
Why are we using the loader's fetch here?

Reply:
same response as below. i can add this but it feels leaky...